pkg/maps/metricsmap/metricsmap.go

```diff
@@ -86,6 +133,64 @@ var (
 		})
 )
 
+// updatePrometheusMetrics checks the metricsmap key value pair
+// and determines which prometheus metrics along with respective labels
+// need to be updated.
+func updatePrometheusMetrics(key *Key, val *Value) {
+	var counter prometheus.Counter
+	var err error
+	if key.IsDrop() {
+		counter, err = metrics.DropCount.GetMetricWithLabelValues(key.DropForwardReason(), key.Direction())
+	} else {
+		counter, err = metrics.ForwardCount.GetMetricWithLabelValues(key.Direction())
+	}
+	if err != nil {
+		log.WithError(err).Warn("failed to update prometheus metrics")
+		return
+	}
+	oldValue := metrics.GetCounterValue(counter)
```

Comment:
How does this behave across agent restarts? What will be the value of the counter after an agent restart?

Reply:
This persist across agent restarts (Since it is a counter and Prometheus will never reset it)
The whole point of using a counter was to avoid being reset to zero on agent restart.
@rlenglet

Reply:
@tgraf : My understanding is that the BPF Maps will persist across restarts. If the in-memory prometheus state is lost due to agent restart, we will simply update it with the value in the BPF maps? Is that the concern?

Reply:
The BPF map is persistent, I assume that the in memory Prometheus counter is not. So we will `Add()` the full delta between `0` and the value in the bpf map to the counter each time the agent restarts.

It would be more accurate to not use a counter for now and just sync the BPF map value to get started.

Reply:
How is the logic in this PR not accurate?
"Adding a value to the counter" only modifies the counter's in-memory value. And that in-memory value is sampled periodically by Prometheus. "Adding a value to the counter" is not sending anything to Prometheus.

Reply:
The missing piece of information was this: `Any time a counter appears to decrease it’ll be treated as though there was a reset to 0 right after the first data point.` This resolves the mystery for me how this was able to handle both BPF map resets and agent restarts with persistent maps.